Sources/SPTDataLoader/SPTDataLoaderService.m

```diff
@@ -378,12 +468,17 @@ - (void)URLSession:(NSURLSession *)session
         // No-op
         // Use default handing
     }
-    
+
     completionHandler(disposition, credential);
 }
 
 #pragma mark NSURLSessionTaskDelegate
 
+- (void)URLSession:(NSURLSession *)session didBecomeInvalidWithError:(nullable NSError *)error
+{
+    [_invalidatedSessions addObject:session];
```

Comment:
As an option, we can keep track of invalidated sessions without extra set: a Service owns a SessionSelector, which in turn manages only two (or rather, up to two) session instances, so I guess we could leverage that instead.
Although I'm wondering if a Service should even start performing a request if its only Session has been invalidated. Wdyt?

Reply:
Not sure if I'm following why that would be better. This DataLoaderService is the one implementing `NSURLSessionTaskDelegate`, so if we were to go on that path this service would still have to tell SessionSelector which session is invalidated and then retrieve it after. It seems an unnecessary indirection 🤔 .

About not making a request on an invalidated session, Gianni made a change recently that prevents that from happening on a retry - but we'd still like to know if this happens in a regular request (i.e. leave breadcrumb and crash)